check.ml

```diff
@@ -0,0 +1,123 @@
+open Instr
+
+(* Those are exception at the scope of the program *)
+exception MissingMain
+exception InvalidMain
+exception DuplicateFunctionDeclaration of identifier
+exception DuplicateVersion of identifier * label
+exception EmptyFunction of identifier
+exception DuplicateParameter of identifier * variable
+
+(* The following are exception at the scope of a particular
```

Comment:
Nitpicking typos: The following are exception**s** ... they will be wrapped in a**n** ErrorAt exception.